### PR TITLE
fix: bash script to utilise dynamic input

### PR DESCRIPTION
**Pull request by kushthedude**
Signed-off-by: Kush Trivedi <kushthedude@gmail.com>

**Description**

This PR refers #1872

### CODE REVIEW

**Comment by kushthedude** [comment]
With this, we can use AKS cluster in mesheryctl 🎉

File: mesheryctl/pkg/utils/scripts.go [hunk 1]
```diff
@@ -141,20 +145,16 @@ func GenerateConfigAKS() error {
...
 	
 	fetch_aks_script() {
-		echo -n "enter your resource group"
-		printf "\n"
-		read -r resource_group
-		echo -n "enter your aks cluster name"
 		printf "\n"
-		read -r aks_name
-		az aks get-credentials --resource-group "${resource_group}" --name "${aks_name}" --file "${TARGET_FILE}"
+		az aks get-credentials --resource-group "${Resource_Group}" --name "${Cluster_Name}" --file "${TARGET_FILE}"
```
Comment (leecalcote) [inline mesheryctl/pkg/utils/scripts.go:149]: Where are we checking for the `az` dependency?

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: mesheryctl/pkg/utils/scripts.go (hunk 1 above)
Comment (kushthedude) [reply mesheryctl/pkg/utils/scripts.go:149]: We are not as it is a prereq, or do we want to make script fetch the az too if not found?

Comment (leecalcote) [reply mesheryctl/pkg/utils/scripts.go:149]: Great question. For now, if we just gracefully handle the potentially missing prerequisite error, we can merge this PR and put out a nice-to-have PR for a contributor to consider picking up two auto install `az`￼￼￼.

Comment (kushthedude) [reply mesheryctl/pkg/utils/scripts.go]: Done!

Comment (leecalcote) [reply mesheryctl/pkg/utils/scripts.go:156]: I wonder if this is the same formatting issue being run into by Prateek T.?

Comment (kushthedude) [reply mesheryctl/pkg/utils/scripts.go:156]: Yes, the formatting issue is due to the presence of editor config for golang files ig. Every IDE formats according to golang standards but after seeing editorconfig the presets are overridden resulting in failure of gofmt. 😓

File: mesheryctl/pkg/utils/scripts.go [hunk 2]
```diff
@@ -1,172 +1,172 @@
...
+	set -e
+	set -o pipefail
+
+	TARGET_FOLDER="/tmp/meshery"
+	TARGET_FILE="$TARGET_FOLDER/kubeconfig.yaml"
+
+	Resource_Group="%s"
+	Cluster_Name="%s"
+
+	check_for_az() {
```
Comment (leecalcote) [inline mesheryctl/pkg/utils/scripts.go:141]: Do we exit if not found? or is this handled elsewhere?

File: mesheryctl/pkg/utils/scripts.go (hunk 1 above)
Comment (leecalcote) [reply mesheryctl/pkg/utils/scripts.go:156]: Got it, ok. Prateek's PR was held up on account of this perhaps for longer than it should have been in retrospect.

File: mesheryctl/pkg/utils/scripts.go (hunk 2 above)
Comment (leecalcote) [reply mesheryctl/pkg/utils/scripts.go:141]: `az version` will error with a "command not found", I assume, but no instructions to users to install the prerequisite.

Comment (kushthedude) [reply mesheryctl/pkg/utils/scripts.go:141]: We exit if the command is not found.

Comment (kushthedude) [reply mesheryctl/pkg/utils/scripts.go:141]: > `az version` will error with a "command not found", I assume, but no instructions to users to install the prerequisite.

I thought of handling this later by providing an option to install `az` in the mesheryctl utility itself

Comment (leecalcote) [reply mesheryctl/pkg/utils/scripts.go:141]: Makes sense, however, in effect, this check doesn't do anything different than if the check wasn't included.

Comment (kushthedude) [reply mesheryctl/pkg/utils/scripts.go:141]: done the changes, see if it looks alright!

**Comment by leecalcote** [comment]
Bugs to fix:

- https://github.com/layer5io/meshery/issues/1896
- https://github.com/layer5io/meshery/issues/1895

**Pull request merged**